docs/start/getting-started/fragments/flutter/add-api.md

```diff
@@ -1,65 +1,143 @@
 
-    ```
-    ? Select an Analytics provider (Use arrow keys)
-        `Amazon Pinpoint`
-    ? Provide your pinpoint resource name: 
-        `yourPinpointResourceName`
-    ? Apps need authorization to send analytics events. Do you want to allow guests and unauthenticated users to send analytics events? (we recommend you allow this when getting started) 
-        `Yes`
-    ```
+```
+amplify pull --appId <appId> —envName staging
 
-5. To save all your changes and to create your AWS resources, run the following command last:
+? Choose your default editor:
+    `<your editor of choice>`
+? Choose the type of app that you're building
+    `flutter`
+? Where do you want to store your configuration file?
+    `./lib/`
+? Do you plan on modifying this backend?
+    `Yes`
+```
 
-    ``` 
-    amplify push 
-    ```
+## Add authentication
 
-    After these steps, you should notice a `amplifyconfiguration.dart` file within your lib directory of your project.  Guard this file carefully!  It contains sensitive information that your app will use to establish a secure communication with your backend AWS resources.  If it is lost or corrupted, you can always regenerate it by repeating the above steps again with the Amplify CLI. 
+Since our **Todo** model specifies an @auth directive, we do need to first add authentication.
 
-At this point you are ready to run your app.  Go back to Android Studio and at the top bar click on the green play button to deploy.
+<amplify-block-switcher>
+
+<amplify-block name="Admin UI">
+
+<inline-fragment src="~/start/getting-started/fragments/flutter/blocks/add-api-add-authentication-admin-ui.md"></inline-fragment>
+
+</amplify-block>
+
+<amplify-block name="CLI">
+
+<inline-fragment src="~/start/getting-started/fragments/flutter/blocks/add-api-add-authentication-cli.md"></inline-fragment>
+
+</amplify-block>
+
+</amplify-block-switcher>
+
+## Add Api and Auth plugins
+
+Now we just need to configure our app with the plugins. Start by modifying `pubspec.yaml` in your project root directory and add the Amplify plugins to the project dependencies
+
+```diff
+  dependencies:
+    flutter:
+      sdk: flutter
+
+    amplify_flutter: <1.0.0
+    amplify_datastore: <1.0.0
++   amplify_api: <1.0.0
++   amplify_auth_cognito: <1.0.0
+```
+
+Install the dependencies by running the following command. Depending on your development environment, you may perform this step via your IDE (or it may even be performed for you automatically).
+
+```bash
+flutter pub get
+```
+
+Update the `main.dart` file and import the new packages.
+
+```diff
+  import 'package:amplify_flutter/amplify.dart';
+  import 'package:amplify_datastore/amplify_datastore.dart';
++ import 'package:amplify_api/amplify_api.dart';
++ import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+```
+
+Declare the additional plugins at the top of the `_TodosPageState` class.
```

Comment:
For these type of direct / single sentence instructions, another sentence just to get the customer aware of _why_ this is needed could be a nice addition. Something like:

> Declare the additional plugins at the top of the `_TodosPageState` class. This will ensure DataStore has access to the AppSync API.

Reply:
I tried to look these over and I think we can elaborate on the "why" at the top of this section without actually adding too much per small instruction simply because they're all just doing the same thing for the same reason. I.e. import, declare and add plugins in order to ____

I also don't want to introduce the concept of AppSync at all since a customer can realistically get through this tutorial without ever understanding that AppSync is behind some of the magic.

I will update this section to elaborate a bit on why we're doing this though!